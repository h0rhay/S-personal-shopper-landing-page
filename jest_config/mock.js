export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
