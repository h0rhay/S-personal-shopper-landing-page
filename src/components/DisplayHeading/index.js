import PropTypes from 'prop-types';
import React from 'react';
import styled, { css } from 'styled-components';

const DisplayHeadingStyles = styled.h1`
  width: 100%;
  text-align: center;
  position: absolute;
  top: 25vmin;
  @media (max-width: 812px) {
    top: 90vmin;
  }
  @media (min-width: 767px) and (max-width: 812px) {
    top: 92vmin;
  }
  word-wrap: break-word;
  ${({ theme }) =>
    theme.themeType === 'dark'
      ? css`
          text-shadow: 0px 2px 7px ${theme.shadow};
        `
      : css`
          text-shadow: -1px 1px 0 ${theme.shadow}, 1px 1px 0 ${theme.shadow}, 1px -1px 0 ${theme.shadow},
            -1px -1px 0 ${theme.shadow};
        `}
  letter-spacing: 20px;
  @media (max-width: 812px) {
    letter-spacing: 10px;
  }
  font-size: 6vmax;
  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      ${theme.typography.mobile.displayHeading};
    } ;
  `};
`;

const DisplayHeading = ({ children, ...delegated }) => (
  <DisplayHeadingStyles {...delegated}>{children}</DisplayHeadingStyles>
);

DisplayHeading.propTypes = {
  children: PropTypes.string,
};

export default DisplayHeading;
