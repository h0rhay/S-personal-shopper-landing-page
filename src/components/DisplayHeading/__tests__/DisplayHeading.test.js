import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import DisplayHeading from '../index';

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <DisplayHeading />
    </ThemeProvider>
  );

describe('DisplayHeading component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme });
    expect(container).toMatchSnapshot();
  });
});
