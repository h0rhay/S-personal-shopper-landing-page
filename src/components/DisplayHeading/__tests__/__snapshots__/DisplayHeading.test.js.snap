// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayHeading component should match screenshot 1`] = `
<div>
  <h1
    class="sc-bdvvtL fFpqFh"
  />
</div>
`;
